weakmw.tex

```diff
@@ -3,136 +3,171 @@ \chapter{The Weak Mordell-Weil Theorem}\label{ch:weakmw}
+	of $\frac{1}{\alpha}\O_{K}$ is $\p_i^{-m}$ so by construction
+	$\frac{1}{\alpha}\in\O_{K,S}.$ Since
+	$\alpha\in \left(\p_i\O_{K,S}\right)^m$ this shows $(\p_i\O_{K,S})^m$
+	is the unit ideal. It follows from the unique factorization of ideals
+	in the Dedekind domain $\O_{K,S}$ that $\p_i\O_{K,S}$ is the unit ideal.
+	
+	Now we can show $\O_{K,S}$ is a principal ideal domain. Let $\P$
+	be a prime ideal of $\O_{K,S}$. Since the $\p_i$ generate
+	the class group of $\O_K$, the restriction of $\P$ to $\O_K$ is
+	equivalent modulo a principal ideal to a product of the primes
+	$\p_i$. Therefore $\P$ is equivalent modulo a principal ideal
+	to a product of ideals of the form $\p_i\O_{K,S}$. Because we showed
+	$\p_i\O_{K,S}$ was the unit ideal, this means $\P$ is principal.
 	
 	Next enlarge $S$ so that all primes over $n\O_K$ are in~$S$.
 	Note that $\O_{K,S}$ is still a PID.  Let
 	$$
 	 K(S,n) =
 	 	\{
-	 	a \in K^*/(K^*)^n : n \mid \ord_\p(a) \text{ for all } \p\not\in S
+	 	a \in K^*/(K^*)^n \colon n \mid \ord_\p(a) \text{ for all } \p\not\in S
 	 	\}.
 	$$
 	Then a refinement of the arguments at the beginning of
 	this section show that $L$ is generated by all $n$th roots
-	of the elements of $K(S,n)$.  It thus suffices to prove
-	that $K(S,n)$ is finite.
+	of the elements of $K(S,n)$ (specifically, their representatives in $K$).
+	Thus it suffices to prove that $K(S,n)$ is finite.
 	
-	There is a natural map
+	If $a \in \O_{K,S}^*$ then $\ord_\p(a)=0$ for all $\p\notin S$.
+	So there is a natural map
 	$$
-	\phi: \O_{K,S}^* \to K(S,n).
+		\phi: \O_{K,S}^* \to K(S,n)
 	$$
+	sending $a$ to it's residue class in $K^*/(K^*)^n$.
 	Suppose $a\in K^*$ is a representative of an element in $K(S,n)$.
 	The ideal $a \O_{K,S}$ has a factorization which is a product of $n$th
 	powers, so it is an $n$th power of an ideal. Since $\O_{K,S}$ is a PID,
 	there is $b\in \O_{K,S}$ and $u\in \O_{K,S}^*$ such that
 	$$
-	   a = b^n \cdot u.
+		a = b^n \cdot u.
 	$$
-	Thus $u\in \O_{K,S}^*$ maps to $[a] \in K(S,n)$.  Thus $\phi$
+	Thus $u\in \O_{K,S}^*$ maps to $[a] \in K(S,n)$. This shows $\phi$
 	is surjective.
 	
-	Recall that we proved {\em Dirichlet's unit theorem} (see
-	Theorem~\ref{thm:units}), which asserts that the group $\O_K^*$ is a
-	finitely generated abelian group of rank $r+s-1$.  More generally, we
+	Recall {\em Dirichlet's unit theorem} (Theorem~\ref{thm:units}),
+	which asserts that the group $\O_K^*$ is a finitely generated
+	abelian group of rank $r+s-1$.  More generally, we
 	now show that $\O_{K,S}^*$ is a finitely generated abelian group of
-	rank $r+s+\#S -1$.  Once we have shown this, then since $K(S,n)$ is torsion
-	group that is a quotient of a finitely generated group, we will conclude
-	that $K(S,n)$ is finite,
-	which will prove the theorem.
-	%TODO what is this?
-	\index{$S$-unit theorem}\index{Dirichlet!$S$-unit theorem}
+	rank $r+s+\#S -1$.
+	This result with above would imply $K(S,n)$ is a torsion group
```

Comment:
reword -- just make really easy to understand which thing above, which result, etc.